src/main/kotlin/no/nav/syfo/aktivitetskrav/domain/AktivitetskravVurdering.kt

```diff
@@ -70,31 +56,30 @@ data class AktivitetskravVurdering private constructor(
                 beskrivelse = beskrivelse,
                 arsaker = arsaker,
                 frist = frist,
-            )
+            ).also { it.validate() }
         }
+    }
+}
 
-        private fun validate(
-            status: AktivitetskravStatus,
-            arsaker: List<VurderingArsak>,
-        ) {
-            if (status !in allowedVurderingStatus) {
-                throw IllegalArgumentException("Can't create vurdering with status $status")
-            }
-            if (!status.requiresVurderingArsak() && arsaker.isNotEmpty()) {
-                throw IllegalArgumentException("$status should not have arsak")
-            }
-            if (status.requiresVurderingArsak() && arsaker.isEmpty()) {
-                throw IllegalArgumentException("Must have arsak for status $status")
-            }
-            if (arsaker.any { it.validForStatus != status }) {
-                throw IllegalArgumentException("Must have valid arsak for status $status")
-            }
-        }
+fun AktivitetskravVurdering.validate() {
+    if (!status.isAllowedChangedVurderingStatus()) {
+        throw IllegalArgumentException("Can't create vurdering with status $status")
+    }
+    if (!status.requiresVurderingArsak() && arsaker.isNotEmpty()) {
```

Comment:
Litt på siden av dene PR'en, men: Hvis det ikke er lov å sende med årsak hvis det ikke er required blir dette navnet litt misvisende?
`not required` == `valgfritt`
`not required` != `ikke lov`

Reply:
Får evnt gjøre noe med dette i en egen PR.